util/movement-types/src/lib.rs

```diff
@@ -118,27 +159,55 @@ impl Block {
 		Self { metadata, parent, transactions, id }
 	}
 
-	pub fn id(&self) -> Id {
-		self.id.clone()
+	pub fn into_parts(self) -> (BlockMetadata, Vec<u8>, BTreeSet<Transaction>, Id) {
+		(self.metadata, self.parent, self.transactions, self.id)
+	}
+
+	pub fn id(&self) -> &Id {
+		&self.id
+	}
+
+	pub fn parent(&self) -> &Vec<u8> {
```

Comment:
This is questionable in multiple ways now.
At the surface level, this should return `&[u8]`.
But how does a byte array represent a parent? Is it [supposed to be](https://github.com/movementlabsxyz/movement/blob/31386f74a2700920fec093775c221acdc9c6c7a2/protocol-units/sequencing/memseq/sequencer/src/lib.rs#L100) an `Id` as well?
I don't see this member used much.

Reply:
It hasn't been used yet. It's better as an `Id`. But, the outstanding question per share-sequencer discussions was whether this might need to be something more than just the hash of the previous block. That is, you might want to do something like...

```rust
pub struct Parent {
    chain: Id,
    block: Id
}
```

...for atomic transaction bundles where shared ordering amongst chains is temporary.

Reply:
Should we have this as an opaque type for now? I'd prefer not to have anything if it's not used, but if it needs to be on-chain, perhaps a newtype wrapping a dummy ID would do for now.

Reply:
Yeah, let's go with wrapping a dummy ID.